modules/mod_articles/src/Helper/ArticlesHelper.php

```diff
@@ -227,16 +228,40 @@ public function getArticles(Registry $params, SiteApplication $app)
                 && $input->get('option') === 'com_content'
                 && $input->get('view') === 'article'
             ) {
-                $articlesList[] = $input->get('id', 0, 'UINT');
+                $articlesList[] = $currentArticleId;
             }
 
             $articlesListToProcess = $params->get('excluded_articles', '');
         }
 
         foreach (ArrayHelper::fromObject($articlesListToProcess) as $article) {
+            if (
+                $ex_or_include_articles === 1
+                && $params->get('exclude_current', 1) === 1
+                && $input->get('option') === 'com_content'
+                && $input->get('view') === 'article'
```

Comment:
```suggestion
                $ex_or_include_articles === 1
                && $params->get('exclude_current', 1) === 1
                && $input->get('option') === 'com_content'
                && $input->get('view') === 'article'
```
These checks can be assigned to a variable above the loop to be executed once.  Then no need to do the same checks except with the id in the loop. This variable can be used again in the next if statement.

Reply:
For now I will let it as it is, we need to get it in